pandas/core/ops.py

```diff
@@ -990,6 +991,93 @@ def _construct_divmod_result(left, result, index, name, dtype):
     )
 
 
+def dispatch_to_extension_op(left, right, op_name=None, is_logical=False):
+    """
+    Assume that left is a Series backed by an ExtensionArray,
+    apply the operator defined by op_name.
+    """
+
+    method = getattr(left.values, op_name, None)
+    deflen = len(left)
+    excons = type(left.values)._from_sequence
+    exclass = type(left.values)
+    testseq = left.values
+
+    if is_logical:
+        if exclass._logical_result is not None:
+            excons = exclass._logical_result
+        else:
+            excons = None  # Indicates boolean
+
+    # The idea here is as follows.  First we see if the op is
+    # defined in the ExtensionArray subclass, and returns a
+    # result that is not NotImplemented.  If so, we use that
+    # result. If that fails, then we try an
+    # element by element operator, invoking the operator
+    # on each element
+
+    # First see if the extension array object supports the op
+    res = NotImplemented
+    if method is not None and inspect.ismethod(method):
+        rvalues = right
+        if is_extension_array_dtype(right) and isinstance(right, ABCSeries):
+            rvalues = right.values
+        try:
+            res = method(rvalues)
+        except TypeError:
+            pass
+        except Exception as e:
+            raise e
```

Comment:
Why not doing `op(left.values, right/right.values)`? What does this manual checking/trying does that the former does not?